drift.py

```diff
@@ -1,146 +1,90 @@
+from spack import *
+from spack.cmd.diff import compare_specs
 
 # Define SUCCESS for comparing command return codes.
 SUCCESS = 0
-args = None
-
+# Define result dataclass
+@dataclass
 class Result:
-    abstract:str
+    name:str
+    version:str
     commit:str
     tags:list
     timestamp:str
-    spec:str
-    version:str
-
-    def __init__(self):
-        self.abstract = ""
-        self.commit = ""
-        self.tags = []
-        self.timestamp = ""
-        self.spec = ""
-        self.version = ""
-
 
 class DriftAnalysis(Helicase):
-    specs = []
-    last = {}
-    def __init__(self, specs):
-        self.specs = specs
-        for spec in specs:
-            self.last[spec] = ""
+    def __init__(self, specs=None):
+        self.last = {}
+        self.specs = specs or []
 
     def analyze(self, commit):
-        for spec in self.specs:
-            out = spack("spec --json " + spec)
-            verToken = spec.find("@")
-            if verToken < 0: verToken = len(spec)
-            name = spec[:verToken]
-            result = Result()
-            result.commit = commit.hash
-            result.abstract = name
-            result.timestamp = str(commit.author_date)
-            if len(spec[verToken:]) > 0:
-                result.version = spec[verToken+1:]
-            if out.returncode == SUCCESS:
-                concrete_spec = json.loads(out.stdout)["spec"]
-                # In the case that the concretized spec hash changes
-                # we want to record the commit where this happens.
-                if self.last[spec] != "" and self.last[spec][0][name]["full_hash"] != concrete_spec[0][name]["full_hash"]:
-                    # Check if the package version is different.
-                    if self.last[spec][0][name]["version"] != concrete_spec[0][name]["version"]:
-                        result.tags += ["pkg-updated"]
-                    # +---------------------+
-                    # |     Dependencies    |
-                    # +---------------------+
-                    result.tags += set(self.tag_deps(concrete_spec, spec, name))
-                    # +---------------------+
-                    # |      Variants       |
-                    # +---------------------+
-                    result.tags += set(self.tag_variants(concrete_spec, spec, name))
-                    # Send Inflection Point
-                    send(result)
-                # Save concrete spec hash as last hash.
-                self.last[spec] = concrete_spec
-            else:
-                # If the spec doesn't concretize properly we also want
-                # to record the commit at which this occurred.
-                result.tags += ["concretizaton-failed"]
+        for abstract_spec in self.specs:
+            # Separate out spec version and name.
+            name = abstract_spec
+            version = ""
+            if "@" in abstract_spec:
+                name, version = abstract_spec.split("@", 1)
```

Comment:
This assumes your spec is just `"name@version"`, but there is an entire grammar to parse here.  I would do:

```python
for spec_string in self.specs:
    abstract_spec = Spec(spec_string)
```

Then you can refer to `spec.name`, `spec.version`, etc.